packages/wrangler/src/triggers/deploy.ts

```diff
@@ -310,6 +312,129 @@ export function getSubdomainValues(
 	};
 }
 
+async function validateSubdomainMixedState(
+	props: Props,
+	accountId: string,
+	scriptName: string,
+	configFlags: { workers_dev?: boolean; preview_urls?: boolean },
+	desired: { workers_dev: boolean; preview_urls: boolean },
+	firstDeploy: boolean
+): Promise<{
+	workers_dev: boolean;
+	preview_urls: boolean;
+}> {
+	const { config } = props;
+
+	// Early return if check disabled through environment variable.
+	if (
+		["true", "1", "yes", "on"].includes(
+			process.env.WRANGLER_DISABLE_SUBDOMAIN_MIXED_STATE_CHECK?.toLowerCase() ??
```

Comment:
Can we move this to `wrangler/src/environment-variables`?
Mostly so that we have all potential env-vars declared in a single place?

Reply:
Also we have `getBooleanEnvironmentVariableFactory()` which keeps our approach to boolean environment variables consistent.